src/DocBlock/TypeExpression.php

```diff
@@ -38,143 +38,151 @@ final class TypeExpression
      *
      * @internal
      */
-    public const REGEX_TYPES = '(?<types>(?x) # one or several types separated by `|` or `&`
-'.self::REGEX_TYPE.'
-        (?:
-            \h*(?<glue>[|&])\h*
-            (?&type)
-        )*+
-    )';
-
-    private const REGEX_TYPE = '(?<type>(?x) # single type
-            (?<nullable>\??\h*)
-            (?:
-                (?<array_shape>
-                    (?<array_shape_start>(?i)(?:array|list|object)(?-i)\h*\{\h*)
-                    (?<array_shape_inners>
-                        (?<array_shape_inner>
-                            (?<array_shape_inner_key>(?:(?&constant)|(?&identifier))\h*\??\h*:\h*|)
-                            (?<array_shape_inner_value>(?&types_inner))
+    public const REGEX_TYPES = <<<'EOD'
+        (?<types>(?x) # one or several types separated by `|` or `&`
+
+        EOD.self::REGEX_TYPE.<<<'EOD'
```

Comment:
This looks a bit weird to me, I think I prefer the regular single-quoted string version.